tests/harness/go/harness.go

```diff
@@ -21,13 +21,11 @@ func main() {
 	da := new(ptypes.DynamicAny)
 	checkErr(ptypes.UnmarshalAny(tc.Message, da))
 
-	if msg, ok := da.Message.(interface {
+	msg := da.Message.(interface {
 		Validate() error
-	}); ok {
-		checkValid(msg.Validate())
-	} else {
-		checkValid(nil)
-	}
+	})
```

Comment:
This will panic if the message does not implement the interface. The original intent here was that a message not implementing the interface should be considered valid as if it had no validation rules, but since the purpose of this is to test the PGV generation, it seems unnecessary. This can be changed to:

```go
if msg, ok := da.Message(interface{/*...*/}); ok {
  checkValid(msg.Validate())
} else {
  checkValid(errors.New("does not implement Validate interface"))
}
```

or similar

Reply:
That was exactly the intent. The panic produces essentially the same error message and the same behavior from the test harness process (it sees that a child process exited with a non-zero status code) so I didn't see the need to make it explicit. I think a comment would suffice but don't have strong feeling either way.